pushnotifications/src/main/java/com/pusher/pushnotifications/internal/ServerSyncJob.kt

```diff
@@ -1,35 +1,44 @@
 package com.pusher.pushnotifications.internal
 
 import com.pusher.pushnotifications.api.DeviceMetadata
-import com.squareup.moshi.adapters.PolymorphicJsonAdapterFactory
+import com.squareup.moshi.JsonClass
+import dev.zacsweers.moshisealed.annotations.TypeLabel
 import java.io.Serializable
 
-sealed class ServerSyncJob: Serializable {
-
-    companion object {
-        // the following is required for the Persistent Job Queue to read and write data correctly.
-        // if you add another ServerSyncJob, you must also add it below here too!
-        val jsonAdapterFactory: PolymorphicJsonAdapterFactory<ServerSyncJob>
-                = PolymorphicJsonAdapterFactory.of(ServerSyncJob::class.java, "ServerSyncJob")
-                .withSubtype(StartJob::class.java, "StartJob")
-                .withSubtype(RefreshTokenJob::class.java, "RefreshTokenJob")
-                .withSubtype(SubscribeJob::class.java, "SubscribeJob")
-                .withSubtype(UnsubscribeJob::class.java, "UnsubscribeJob")
-                .withSubtype(SetSubscriptionsJob::class.java, "SetSubscriptionsJob")
-                .withSubtype(ApplicationStartJob::class.java, "ApplicationStartJob")
-                .withSubtype(SetUserIdJob::class.java, "SetUserIdJob")
-                .withSubtype(StopJob::class.java, "StopJob")
-    }
-}
+@JsonClass(generateAdapter = true, generator = "sealed:ServerSyncJob")
```

Comment:
It does not seem like we're using code gen here (no kapt dependency). Those annotations here look redundant.

Reply:
Or rather I hope that all that boiler-plate is redundant ;) Definitely the annotation values about generation seem redundant.

Reply:
Yeah, I think you were right! I removed almost all of them and everything works correctly. I think the one on the sealed type is still necessary as it tells everything else it's a sealed type. Does that make sense?

Reply:
Awesome, thanks Danielle! I've checked that making the annotation that needs to stay not conveying information that code gen is involved (whilst it isn't), works (at least the tests involved in this diff are green on my machine).
```suggestion
@JsonClass(generateAdapter = false, generator = "sealed:ServerSyncJob")
```
Well it still sounds a bit like code gen is involved ("generator") but not that explicitly and hopefully `generateAdapter = false` clearly conveys that it is not. It feels though like an imperfection of that super early preview of the library extension so that's probably the best what we can achieve now.

Reply:
That sounds fair! I've my code to reflect that.